### DISCUSSION TITLE
Json to std::function (lambda)

### DISCUSSION THREAD

**Discussion by carul17** [nlohmann/json#3821] [Q&A] [answered]
Lets say I have these classes:

```
class Player{
      int rank;
      int score;
};
class Villain{
     int rank;
     int score;
};
class boss{
int rank;
int score;
};

```
and I have this json file:

```
{
"condition": "player.rank >= villain.rank"
"condition": "player.score >= villain.score"
}
```

The lambda in the first case should be something like this:
```
Player p;
Villain v;
auto condition = [&p, &v](int rank){ return if(player.rank >= villain.rank};
```

I want to turn the information into this lambda to pass to a function. I will not know what the condition will be and further instead of 'Villain', the json information may contain 'Boss' which is another similar type I have. The json may have many different conditions similar to this and I want to be able to convert it into a lambda without knowing the information beforehand.

I will pass this to a function which will handle the condition like this:

```
void handleCondition(std::function<bool(int)> condition){
       if(condition)
            //do something
}
```

How do I convert the json information into this lambda? Or is there a better way to do this instead of turning it into a lambda?

**Comment by falbrechtskirchinger** [comment, marked as answer]
You'll have to implement your own parser and a minimal interpreter or integrate any scripting engine of your choice.
Here are two options:
https://github.com/ThePhD/sol2
https://github.com/ChaiScript/ChaiScript

It all depends on the complexity of the conditions you want to support.

**Reply by carul17** [reply]
Thank you!
